activemodel/lib/active_model/dirty.rb

```diff
@@ -180,7 +180,7 @@ def attribute_changed?(attr, options = nil) #:nodoc:
         result &&= options[:to] == __send__(attr) if options.key?(:to)
         result &&= options[:from] == changed_attributes[attr] if options.key?(:from)
       end
-      result
+      !!result
```

Comment:
@jeremy The change is trying to fix this issue https://github.com/rails/rails/issues/24220. The issue is caused by the fact that method `attribute_changed?` could return nil. I have proposed a more comprehensive fix https://github.com/rails/rails/pull/24242, but it has been considered going too far because some changes are not desired. I also tried to fix it by using `result.present?`, but this solution is also not preferable(@pixeltrix left some comments here for that https://github.com/rails/rails/pull/24242).